src/shared/lib/sortedFindWith.ts

```diff
@@ -0,0 +1,31 @@
+// Use binary search to find the element in `sortedArray` for which `oracle` returns 0.
+// It's assumed that sortedArray is sorted, i.e. `sortedArray.map(oracle) = [-1,-1,...,-1,0,1,1,...,1]`
+// Behavior undefined on non-sorted arrays.
+
+export function sortedFindWith<T>(
```

Comment:
i don't see where this is being used in code.  when i search for "sortedFindWith" i only get this file and spec file.

Reply:
I had originally written it for changes which I then removed. But I thought it was a useful function, and I wrote it and tested it, so I decided to leave it in.